python/ray/air/config.py

```diff
@@ -187,7 +187,8 @@ def _resources_per_worker_not_none(self):
         resources_per_worker = {
             k: v for k, v in self.resources_per_worker.items() if v != 0
         }
-        resources_per_worker.setdefault("GPU", int(self.use_gpu))
```

Comment:
If use_gpu=False, we will return a resource dictionary of `{"GPU": 0}`, which is still empty.